instrumentation/redis/lib/opentelemetry/instrumentation/redis/utils.rb

```diff
@@ -17,24 +17,46 @@ module Utils
 
         def format_span_name(commands)
           commands.map do |command|
-            format_arg(resolve_command_args(command).first)
+            format_cmd(resolve_command_args(command).first)
           end.join(' ')
         end
 
         def format_statement(command_args)
           command_args = resolve_command_args(command_args)
           return 'AUTH ?' if auth_command?(command_args)
 
-          cmd = command_args.map { |x| format_arg(x) }.join(' ')
-          OpenTelemetry::Common::Utilities.truncate(cmd, CMD_MAX_LEN)
+          cmd = format_cmd(command_args.first)
+          args = command_args[1..-1].map { |arg| format_arg(arg) }
+          statement = args.unshift(cmd).join(' ')
+          OpenTelemetry::Common::Utilities.truncate(statement, CMD_MAX_LEN)
+        end
+
+        def format_cmd(cmd)
+          format_string do
+            cmd.to_s.upcase
+          end
         end
 
         def format_statements(commands)
           commands.map { |command| format_statement(command) }.join("\n")
         end
 
         def format_arg(arg)
-          str = arg.is_a?(Symbol) ? arg.to_s.upcase : arg.to_s
+          format_string do
+            obfuscate_arg(arg.to_s)
+          end
+        end
+
+        private
+
+        def obfuscate_arg(arg)
+          return PLACEHOLDER if config[:enable_statement_obfuscation]
+
+          arg
+        end
```

Comment:
This seems over engineered - I think this is sufficient:
```suggestion
          if config[:enable_statement_obfuscation]
            PLACEHOLDER
          else
            format_string(arg.to_s)
          end
        end

        private
```